### PR TITLE
Add a speech enhancement recipe: egs2/urgent24/enh1

### PR DESCRIPTION
**Pull request by Emrys365**
## What?

This PR adds a speech enhancement recipe for the URGENT 2024 Challenge.

## Why?

This is an official recipe for the URGENT 2024 Challenge.

## See also

https://urgent-challenge.github.io/urgent2024/baseline/

### CODE REVIEW

**Comment by sw005320** [comment]
Will you add a README file?

Also, the CI error happens. Can you check it? (but it does not seem to be related to this PR.)

**Comment by Emrys365** [comment]
Sure. I will add one after I get the baseline result.

File: egs2/urgent24/enh1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    fi
+
+    cd "${cwd}"
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    log "stage 2: Data preparation"
+
+    cd "${odir}"/urgent2024_challenge
+    ./prepare_espnet_data.sh
```
Comment (LiChenda) [inline egs2/urgent24/enh1/local/data.sh:101]: Is this file already in this commit? I can't find it.

File: egs2/urgent24/enh1/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    cd "${cwd}"
+    cp -r "${odir}"/urgent2024_challenge/data "${PWD}/data"
```
Comment (LiChenda) [inline egs2/urgent24/enh1/local/data.sh:103]: Will this `cp` copy a lot of audio files? if so, maybe use `ln` or `mv` will be better.

**Review by LiChenda — COMMENTED** [review]
Hi, @Emrys365 , I left some comments in the data preparation section.

File: egs2/urgent24/enh1/local/data.sh (hunk 1 above)
Comment (Emrys365) [reply egs2/urgent24/enh1/local/data.sh:101]: This file is from the cloned repository. It's created before this stage.

File: egs2/urgent24/enh1/local/data.sh (hunk 2 above)
Comment (Emrys365) [reply egs2/urgent24/enh1/local/data.sh:103]: It should be fine. There are only scp files in the directory.

**Comment by sw005320** [comment]
Thanks, @Emrys365!

**Pull request merged**